### ISSUE TITLE
Problems with desktop file and xdg-open on Linux

### ISSUE THREAD

**Issue by JonnyHaystack** [project-slippi/slippi-desktop-app#24]
The path to the AppImage file in the Exec field of the desktop file is wrapped in quotes, which causes errors such as the following when using xdg-open:
`which: no Slippi-Launcher-1.4.4-x86_64.AppImage" in (./"/home/haystack/source)`

This makes it so you cannot open .slp files from the Slippi site in your browser because the URL handling is done by xdg as well. Opening opening a .slp file from a file browser will also most likely be broken by this although it may depend on your file browser.

If I manually remove the quotes in the Exec field of the desktop file, then it works fine and the slippi:// URLs are handled properly and I can open the files directly with xdg-open as well.

[The spec](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#exec-variables) seems to imply that it should be fine to double quote the path, so I'm not entirely sure why it doesn't work. Could possibly be an upstream bug in xdg-utils.

> Arguments may be quoted in whole. If an argument contains a reserved character the argument must be quoted. The rules for quoting of arguments is also applicable to the executable name or path of the executable program as provided.

For debugging purposes, here's the generated desktop file (`~/.local/share/applications/appimagekit-slippi-desktop-app.desktop`):
```
[Desktop Entry]
Name=Slippi Launcher
Comment=Slippi Desktop App for browsing and playing replays.
Exec=/home/haystack/source/Slippi-Launcher-1.4.4-x86_64.AppImage %U
Terminal=false
Type=Application
Icon=appimagekit-slippi-desktop-app
StartupWMClass=Slippi Launcher
X-AppImage-Version=1.4.4
MimeType=x-scheme-handler/slippi;
Categories=Development;
X-AppImage-BuildId=1K1bOeuJzPhoG9WcseHboZyGhZK
X-Desktop-File-Install-Version=0.24
X-AppImage-Comment=Generated by /tmp/.mount_SlippipSpjHL/AppRun
TryExec=/home/haystack/source/Slippi-Launcher-1.4.4-x86_64.AppImage
```

**Comment by hosaka-corp** [comment]
Weird - I get the same behavior. It turns out `xdg-open` is a shell script (didn't know that!) - maybe something it uses for parsing has changed in some way that breaks things like this. Gonna look into this.

**Comment by JonnyHaystack** [comment]
Yes, looks like they went through the same thought process as me. The spec seems to indicate that it should be fine, but it isn't..

I did a quick experiment to see how spaces in file paths can be handled. Looks like if there's a space you just have to escape it with a backslash, which is the normal alternative to using quotes.

E.g. `Exec=/home/haystack/source/Test\ Things/Slippi-Launcher-1.4.5-x86_64.AppImage %U`

So does electron-builder not have the functionality to perform this escaping when creating desktop files? Admittedly xdg-utils' behaviour is clearly broken, but I'm surprised electron-builder uses quotes when it straight up doesn't even work with xdg-utils.

I think I'm gonna go open an issue upstream with xdg-utils and see how that goes. Seems like they track issues on their GitLab source mirror.

**Comment by hosaka-corp** [comment]
Yeah, I guess ideally `xdg-mime` respects the specification - unless there's some other technical reason why they wouldn't want to support escaping paths with quotes. 

I'll leave this issue open until something happens somewhere upstream.

**Comment by NikhilNarayana** [comment]
https://github.com/electron-userland/electron-builder/issues/2759#issuecomment-504949732

This is fixed since we use 22.8.0+ now.
